cockatrice/src/game/player/player.h

```diff
@@ -226,6 +226,13 @@ private slots:
 
     void initSayMenu();
 
+public:
+    enum EventProcessingOption
+    {
+        SKIP_REVEAL_WINDOW = 0x0001
```

Comment:
this all seems rather complicated, why not make this a simple bool? are there any special eventprocessingoptions you have in mind?

Reply:
I mostly made this an enum because adding a `skipRevealWindow` bool param to every `processEvent` down the chain felt ugly and felt like it was mixing two different levels of abstraction. Maybe adding one bool param is fine, but that just encourages us to keep adding more bool params to it in the future. Making it a flag enum makes it cleaner to extend in the future.

There’s probably some more events that we can save processing effort on during replays. I’m looking into skipping tap animations when rewinding. Might also look into if we can do anything about draw arrow events.

Reply:
if the only case of needing things to be skipped is during replays, why not make it a bool similar to "rewindingReplay"

Reply:
Well, more specifically, `SKIP_REVEAL_WINDOWS` is enabled when rewinding **or** when forward skipping but only if it's more than a certain amount. I suppose we could just have a single boolean represent that condition, but in the future we might want something to happen on a different combination of conditions.

For example, I'm currently looking into skipping the tap/untap animation when rewinding. I've found that it looks nicer if we don't skip tap animations when forward skipping. So now I want the `SKIP_TAP_ANIMATIONS` flag to only be active when rewinding and not when forward skipping. If we used a single bool, we wouldn't be able to do that.

An alternative option we could do is, instead of having a flag enum that represents specific event processing flags that are decided at the point of event generation, we could instead pass an object in the signal that contains info about the playback state (replay vs actual game, skipping vs normal playback, rewinding, etc). Then we move the decision-making logic inside `Player`. That way all the conditional logic is contained in `Player` instead of the event generators.

Thoughts?

Reply:
either would work, benefits of keeping stuff in the player is that this'll give more context to the player, perhaps making other things simpler, benefits of keeping things in the replay tab is that it stays more centralized.

Reply:
I think I prefer going with the current option for now. 

Going with the other option requires us to decide on the structure of the object that we're passing, as well as all the relevant states that we might want to represent. We might want to look into option 2 in the future if we end up with repeated conditional logic in the event generation code in multiple places. For now though, I'm mostly working on replay-related stuff, so it's more convenient to keep the logic centralized in the replay code.

Reply:
sounds good